### DISCUSSION TITLE
SSD/NVMe Drive Wear

### DISCUSSION THREAD

**Discussion by LostSoulfly** [blakeblackshear/frigate#4963] [Ask A Question] [answered]
Frigate docs state:
`Frigate saves from the stream with the record role in 10 second segments. These options determine which recording segments are kept for 24/7 recording (but can also affect events).`

Does this mean each camera is constantly recording to disk every 10 seconds, then deleting the recording if it's not needed later? If so, this would introduce a lot of unnecessary wear on everything but mechanical drives.

I'd really like some insight here, because I use an NVMe drive to cache new recordings for faster access and then move them to a mechanical drive after 3 days automatically but I would not be doing that if it's going to cause an excess of writes to my NVMe cache drive.

If I have this correct, is there perhaps a way to write recordings to memory before determining which recordings to keep and them saving them to disk to prevent this?

**Comment by NickM-27** [comment, marked as answer]
> Does this mean each camera is constantly recording to disk every 10 seconds, then deleting the recording if it's not needed later? If so, this would introduce a lot of unnecessary wear on everything but mechanical drives.

No, frigate writes the segments to cache (RAM) and only moves them to disk if they are to be retained according to the retention settings that are setup.

**Reply by LostSoulfly** [reply]
You're too quick today, Nick. Thanks again

**Reply by NickM-27** [reply]
I will write a PR to update the docs to clarify this further

**Comment by LaurenceGough** [comment]
Great idea on the SSD to cache new recordings, I'd really like to do the same, and as you are not constantly writing to it the wear should be really low. Bradley may I ask how you have this setup? Do you have a file copy setup on a schedule / cron job or something?

**Comment by LostSoulfly** [comment]
I use Unraid and the Mover plugin. All new files are written to an NVMe BTRFS cache pool and then offloaded after it's filled up past a certain point automatically or a file has been in the cache for a few days.

I do not use RAID but rather a parity drive that is calculated whenever files are added/deleted/modified on the fly, so I also gain a bit of extra read performance by having Unraid spread the recordings among my different disks based on free space.

![image](https://user-images.githubusercontent.com/9120519/211653036-f5b91d54-290a-4cee-aa0e-0224ba34dece.png)
![image](https://user-images.githubusercontent.com/9120519/211653299-5e787739-8440-4c86-a46c-052155275401.png)
![image](https://user-images.githubusercontent.com/9120519/211653810-150d2e18-b6ac-4033-a88b-0f02e61439d1.png)

**Reply by LaurenceGough** [reply]
Thanks very much, just a little bit smarter than my setup of one SSD (which the host runs off) and one HDD :D  haha

Nice setup!

**Reply by LostSoulfly** [reply]
Thanks! You got to start somewhere, and I put a lot of thought into my hardware (and it evolved a lot over the years) and I'm pleased with the results.

**Reply by DrSpaldo** [reply]
I was just looking at this situation myself. I currently do the same as you @BradleyShaner, on Unraid with Frigate; the recordings start on the SSD then mover goes to the array when it is filling up. However, I was wondering what kind of difference this makes... IE. how much is stored in RAM? Would it be better going directly to the array to prevent the SSD wear?
